pkg/k8s/service_cache.go

```diff
@@ -550,10 +550,13 @@ func (s *ServiceCache) mergeServiceUpdateLocked(service *serviceStore.ClusterSer
 		s.externalEndpoints[id] = externalEndpoints
 	}
 
+	var oldEPs *Endpoints
```

Comment:
I'd suggest to use `s.correlateEndpoints(id)` to compute the old endpoints. This ensures that we actually propagate all old endpoints, and not only the ones associated with the given remote cluster, so that the difference between new and old endpoints corresponds to the ones that have been removed. The same commend applies to the other occurrence.

Reply:
Got it. Shouldn't it be done in case of local cluster as well? As I can see this is not done in case of local cluster update and delete.

Reply:
Yes, I would uniform that as well. It would also be cool extending the existing tests to assert the correctness of `OldEndpoints`.